src/folio/service-points-api.ts

```diff
@@ -1,15 +1,48 @@
 import FolioAPI from "./folio-api.js"
-import { CqlParams, ServicePoint } from '../schema.js'
+import { ServicePoint, ServicePointDetails, CqlParams } from '../schema.js'
 
 interface ServicePointsResponse {
   servicepoints: ServicePoint[]
 }
 
 export default class ServicePointsAPI extends FolioAPI {
+  // Get an individual service point using the id
+  async getServicePoint(id: string): Promise<ServicePoint> {
+    const servicePoint = await this.get<ServicePoint>(`/service-points/${encodeURIComponent(id)}`)
+    return this.addDetails(servicePoint)
+  }
+
+  // Parse the description string and add the resulting JSON object in the details filed
+  addDetails(servicePoint: ServicePoint) {
+    return {
+      details: parseServicePointDetails(servicePoint),
+      ...servicePoint
+    }
+  }
+
+  // Get a list of all service points along with the details information required for paging
   async getServicePoints(params: Partial<{ params: CqlParams, [key: string]: object | object[] | undefined }>): Promise<ServicePoint[]> {
     const urlParams = this.buildCqlQuery(params)
-
     const response = await this.get<ServicePointsResponse>(`/service-points`, { params: urlParams })
-    return response.servicepoints
+    return  response.servicepoints.map(sp => {
+      return this.addDetails(sp)
+    })
```

Comment:
i think this will do the same thing?
```suggestion
    return  response.servicepoints.map(this.addDetails)
```

Reply:
Yes. Updated.